pkg/generate/app/cmd/newapp.go

```diff
@@ -222,27 +224,58 @@ func (c *AppConfig) SetOpenShiftClient(osclient client.Interface, OriginNamespac
 // AddArguments converts command line arguments into the appropriate bucket based on what they look like
 func (c *AppConfig) AddArguments(args []string) []string {
 	unknown := []string{}
+	c.ArgumentClassificationErrors = map[string]error{}
 	for _, s := range args {
-		switch {
-		case cmdutil.IsEnvironmentArgument(s):
+		if len(s) == 0 {
+			continue
+		}
+
+		validated := false
+
+		// had to move off of elegant switch to capture errors for future display, but mimic
+		// order of precedence established by switch on env arg, then source repo, then comp
+		// ref, then possible template, with no fallthroughs and updates to c.Components or
+		// c.SourceRepositories if a classification of higher precedence succeeds
+
+		rc := cmdutil.IsEnvironmentArgument(s)
+		if rc {
 			glog.V(2).Infof("treating %s as possible environment argument\n", s)
 			c.Environment = append(c.Environment, s)
-		case app.IsPossibleSourceRepository(s):
+			validated = true
+		}
+
+		remote, err := app.IsRemoteRepository(s)
+		c.ArgumentClassificationErrors["Remote access"] = err
+		local, err := app.IsDirectory(s)
+		c.ArgumentClassificationErrors["Local access"] = err
```

Comment:
I was originally that way (whole block, etc .etc.), and then broke slightly from that norm in this sense:
1) The information as to whether the arg satisfied any of the tests seemed useful to me
2) note I don't update the `c.Components` etc. for any of the subsequent classifications on a match has been made ... in that sense I preserve the switch behavior